### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -172,245 +87,121 @@ func currentlyStartedDeviceSerial(alreadyRunningDeviceInfos, currentlyRunningDev
...
-			failf("Failed to update emulator sdk package, error: %s, output: %s", err, out)
-		}
+	var (
+		sdkManagerPath = filepath.Join(cfg.AndroidHome, "tools/bin/sdkmanager")
+		avdManagerPath = filepath.Join(cfg.AndroidHome, "tools/bin/avdmanager")
+		emulatorPath   = filepath.Join(cfg.AndroidHome, "emulator/emulator-headless")
 
-		log.Donef("- Done")
-	}
+		pkg     = fmt.Sprintf("system-images;android-%d;%s;x86", cfg.APILevel, cfg.Tag)
```
Comment (godrei) [inline main.go]: is x86 the only working ABI: https://developer.android.com/ndk/guides/abis?

File: main.go [hunk 2]
```diff
@@ -172,245 +87,121 @@ func currentlyStartedDeviceSerial(alreadyRunningDeviceInfos, currentlyRunningDev
...
+			command.New(emulatorPath, append([]string{
+				"@" + cfg.ID,
+				"-verbose",
+				"-show-kernel",
+				"-no-audio",
+				"-no-boot-anim",
+				"-netdelay", "none",
+				"-no-snapshot",
+				"-wipe-data",
+				"-gpu", "swiftshader_indirect"}, startCustomFlags...)...),
```
Comment (godrei) [inline main.go]: while i was working on this fix, i used some other flags too, would not they help even more:

```
-no-snapshot 
-camera-back none
-camera-front none
-logcat *:e
```

File: step.yml [hunk 3]
```diff
@@ -87,141 +49,28 @@ inputs:
...
-      is_required: false
   - emulator_id: "emulator"
     opts:
-      category: Customize
+      category: Debug
       title: "ID"
       summary: Set the device's ID. (This will be the name under $HOME/.android/avd/)
       description: Set the device's ID. (This will be the name under $HOME/.android/avd/)
-      is_expand: true
       is_required: true
```
Comment (godrei) [inline step.yml]: required input under a category

File: bitrise.yml [hunk 4]
```diff
@@ -1,390 +1,77 @@
 format_version: 1.4.0
 default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
 
-app:
-  envs:
-  - STEP_VERSION: 0.9.2
-
 workflows:
   test:
     before_run:
```
Comment (godrei) [inline bitrise.yml]: pls add `audit-this-step` and our well-known go test steps

File: bitrise.yml [hunk 5]
```diff
@@ -1,390 +1,77 @@
 format_version: 1.4.0
```
Comment (godrei) [inline bitrise.yml]: pls bump the format version

File: step.yml [hunk 6]
```diff
@@ -24,55 +24,17 @@ toolkit:
     package_name: github.com/bitrise-steplib/steps-avd-manager
 run_if: ""
 inputs:
-  - profile: "Nexus 5X"
+  - profile: "pixel"
```
Comment (godrei) [inline step.yml]: the step has invalid `type_tags`, unnecessary `deps`

File: main.go [hunk 7]
```diff
@@ -6,111 +6,26 @@ import (
...
 	"strings"
 	"time"
 
-	"github.com/bitrise-steplib/steps-avd-manager/avdconfig"
-	"github.com/bitrise-tools/go-steputils/input"
-	"github.com/bitrise-tools/go-steputils/tools"
-	shellquote "github.com/kballard/go-shellquote"
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-steputils/tools"
+	"github.com/kballard/go-shellquote"
```
Comment (godrei) [inline main.go]: unnecessary new line

File: step.yml (hunk 3 above)
Comment (trapacska) [reply step.yml]: I left it as it is as this is a required input but has defauklt value so it doesn't requires the user to be confugred to have the step working

File: step.yml [hunk 8]
```diff
@@ -49,6 +44,20 @@ inputs:
       - "android-wear"
       - "android-tv"
       - "default"
+  - abi: "x86"
+    opts:	
+      category: Customize
```
Comment (godrei) [inline step.yml]: pls do not introduce new category

File: step.yml [hunk 9]
```diff
@@ -87,141 +44,42 @@ inputs:
...
-    opts:
-      category: Customize
-      title: "Density"
-      summary: Set the device's display density. This field will override the device profile's default density, if you need the device profile's default density leave this field empty.
-      description: Set the device's display density. This field will override the device profile's default density, if you need the device profile's default density leave this field empty.
-      is_expand: true
-      is_required: false
+  - abi: "x86"
+    opts:	
+      category: Customize
```
Comment (godrei) [inline step.yml]: pls do not introduce a new category for this input

**Pull request merged**